nautobot/utilities/templatetags/helpers.py

```diff
@@ -681,3 +681,16 @@ def custom_branding_or_static(branding_asset, static_asset):
     if settings.BRANDING_FILEPATHS.get(branding_asset):
         return f"{ settings.MEDIA_URL }{ settings.BRANDING_FILEPATHS.get(branding_asset) }"
     return StaticNode.handle_simple(static_asset)
+
+
+@library.filter()
+@register.filter()
+def render_obj_display(model, field_name):
+    """Render the display view of an object."""
+    obj = getattr(model, field_name)
+    if obj:
+        color = getattr(obj, "color", None)
+        display = getattr(model, f"get_{field_name}_display")
+        content = f'<span class="label" style="color: {fgcolor(color)}; background-color: #{color}">{display()}</span>'
```

Comment:
See my [comment](https://github.com/nautobot/nautobot/pull/2822#discussion_r1048841424) on #2822 - we shouldn't need to use `get_{field_name}_display`, should just be able to do `getattr(obj, "name")`